<launch>
	<!--<arg name="map" default="NielsJernesVej14"/>-->
	<arg name="map" default="FBV_7C_L1"/>

	<node pkg="map_server" type="map_server" name="map_server_map"
	      args="$(find navigation)/maps/global_costmap/$(arg map).yaml" output="log"/>
	<node pkg="map_server" type="map_server" name="map_server_localisation"
	      args="$(find navigation)/maps/localisation/$(arg map).yaml" output="log">
		<remap from="map" to="map_localisation"/>
	</node>
	<node pkg="map_server" type="map_server" name="map_server_dangerous_areas"
	      args="$(find navigation)/maps/dangerous_areas/$(arg map).yaml" output="log">
		<remap from="map" to="map_dangerous_areas"/>
	</node>

	<node pkg="amcl" type="amcl" name="amcl" output="log">
		<remap from="map" to="map_localisation"/>
		<remap from="scan" to="lidarScan_merged_tilt_corrected"/>

		<param name="global_frame_id" value="map"/>
		<param name="odom_frame_id" value="odom"/>
		<param name="base_frame_id" value="base_link"/>
		<param name="use_map_topic" value="true"/>
		<param name="first_map_only" value="true"/>
		<param name="tf_broadcast" value="true"/>
		<param name="transform_tolerance" value="0.1"/>
		<param name="save_pose_rate" value="0.5"/>
		<param name="gui_publish_rate" value="-1.0"/>

		<param name="min_particles" value="100"/>
		<param name="max_particles" value="2000"/>
		<param name="kld_err" value="0.01"/>
		<param name="kld_z" value="0.99"/>
		<param name="update_min_d" value="0.2"/>
		<param name="update_min_a" value="0.523598776"/>
		<param name="resample_interval" value="2"/>
		<param name="recovery_alpha_slow" value="0.0"/>
		<param name="recovery_alpha_fast" value="0.0"/>

		<!--<param name="initial_pose_x" value="76.89"/> &lt;!&ndash;njv14&ndash;&gt;-->
		<!--<param name="initial_pose_y" value="12.04"/>-->
		<!--<param name="initial_pose_a" value="0.071436326"/>-->

		<param name="initial_pose_x" value="10.1608"/> <!--fbv7-->
		<param name="initial_pose_y" value="6.6506"/>
		<param name="initial_pose_a" value="-1.51615007"/>

		<param name="initial_cov_xx" value="0.04"/>
		<param name="initial_cov_yy" value="0.04"/>
		<param name="initial_cov_aa" value="0.0171347299"/>

		<param name="laser_model_type" value="likelihood_field"/>
		<param name="laser_min_range" value="-1.0"/>
		<param name="laser_max_range" value="-1.0"/>
		<param name="laser_max_beams" value="20"/>
		<param name="laser_z_hit" value="0.95"/>
		<param name="laser_z_short" value="0.1"/>
		<param name="laser_z_max" value="0.05"/>
		<param name="laser_z_rand" value="0.05"/>
		<param name="laser_sigma_hit" value="0.2"/>
		<param name="laser_lambda_short" value="0.1"/>
		<param name="laser_likelihood_max_dist" value="2.0"/>

		<param name="odom_model_type" value="diff"/>
		<param name="odom_alpha1" value="0.1"/>
		<param name="odom_alpha2" value="0.2"/>
		<param name="odom_alpha3" value="0.2"/>
		<param name="odom_alpha4" value="0.2"/>
		<param name="odom_alpha5" value="0.2"/>
	</node>


	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="log">
		<remap from="cmd_vel" to="mobile_base/commands/velocity"/>

		<rosparam file="$(find navigation)/navigation_params/costmap_common_params.yaml" command="load"
		          ns="global_costmap"/>
		<rosparam file="$(find navigation)/navigation_params/costmap_common_params.yaml" command="load" ns="local_costmap"/>
		<rosparam file="$(find navigation)/navigation_params/local_costmap_params.yaml" command="load"/>
		<rosparam file="$(find navigation)/navigation_params/global_costmap_params.yaml" command="load"/>
		<rosparam file="$(find navigation)/navigation_params/base_local_planner_params.yaml" command="load"/>

		<param name="base_global_planner" value="global_planner/GlobalPlanner"/>
		<param name="GlobalPlanner/use_dijkstra" value="true"/>
		<param name="GlobalPlanner/cost_factor" value="0.55"/>
		<param name="GlobalPlanner/natural_cost" value="66"/>
		<param name="GlobalPlanner/lethal_cost" value="253"/>
		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"/>
		<param name="planner_frequency" value="0.3333"/>
		<param name="planner_patience" value="5.0"/>
		<param name="controller_frequency" value="10"/>
		<param name="controller_patience" value="10.0"/>
	</node>

	<node pkg="navigation" type="costmap_converter_reconfigure" name="costmap_converter_reconfigure" output="log"/>
</launch>
